extensions/m3o/src/avatar.tsx

```diff
@@ -0,0 +1,109 @@
+import { Form, ActionPanel, Action, showToast, Toast, Icon, getPreferenceValues, Clipboard } from "@raycast/api";
+import axios from "axios";
+import fs from "fs";
+import { homedir } from "os";
+
+interface Preferences {
+  apiKey: string;
+}
+
+interface CommandForm {
+  format: string;
+  gender: string;
+  username: string;
+}
+
+export default function Command() {
+  const preferences = getPreferenceValues<Preferences>();
+
+  async function handleSubmit(values: CommandForm) {
+    if (values.format == "") {
+      showToast(Toast.Style.Failure, "Error", "Name is required");
+      return;
+    }
+
+    const toast = await showToast({
+      style: Toast.Style.Animated,
+      title: "Retrieving avatar...",
+    });
+
+    const data = {
+      format: values.format,
+      gender: values.gender,
+      upload: true, // this is so the API returns a URL to the generated avatar
+      username: values.username,
+    };
+
+    const options = {
+      headers: {
+        Authorization: `Bearer ${preferences.apiKey}`,
+      },
+    };
+
+    await axios
+      .post("https://api.m3o.com/v1/avatar/Generate", data, options)
+      .then((response) => {
+        toast.style = Toast.Style.Success;
+        toast.title = "Avatar retrieved successfully";
+        toast.primaryAction = {
```

Comment:
instead of having actions on the toast, we tend to put it in the action panel directly, especially since the toast actions aren't accessible from the keyboard at the moment

Reply:
I added a toast message (as per in my Abstract API extension PR), and hopefully there will be toast action shortcuts in the future.

Reply:
The other PR hasn't been approved yet (cc @pernielsentikaer) and we really like limiting the number of interaction to get a result (1 for the action panel, at least 2, including the mouse) for the toast).

Could you articulate how you see the toast as a better UX than the actions?

Reply:
Hey @mathieudutour, I'll be updating this today to match a better UX experience implemented in the other PR (Abstract API).

Reply:
Sorted.